File: FacebookSrv/controllers/campaign_controller.py
from flask import request, jsonify
from services.facebook_ads_service import create_campaign

def create_campaign_controller():
    data = request.get_json()
    name = data.get('name')
    objective = data.get('objective')

    campaign = create_campaign(name, objective)
    return jsonify(campaign), 201

File: Facebook/src/logics/campaignController.js
const facebookAdsService = require('../services/facebookAdsService');

const createCampaign = async (req, res) => {
  try {
    // Extract campaign data from the request body
    const { name, objective } = req.body;

    // Call the Facebook Ads service to create the campaign
    const campaign = await facebookAdsService.createCampaign(name, objective);

    res.status(201).json(campaign);
  } catch (error) {
    console.error('Error creating campaign:', error);
    res.sendStatus(500);
  }
};

module.exports = {
  createCampaign,
};
